core/src/main/scala/org/apache/spark/scheduler/JobWaiter.scala

```diff
@@ -69,9 +97,11 @@ private[spark] class JobWaiter[T](
   }
 
   def awaitResult(): JobResult = synchronized {
+    Try(_originalHandler = attachSigintHandler())
     while (!_jobFinished) {
       this.wait()
     }
+    Try(detachSigintHandler())
```

Comment:
@davies thank you for reviewing!

My intention for `Try` was to tolerate any error while detaching signal handler since this is on a best effort basis. For eg, if `attachSignalHandler` fails for some reason, `detachSignalHanndler` will fail too.

I rewrote this code using `Try`, `Success`, and `Failure` as follows-
```
  val attachTry = Try(attachSigintHandler())
  while (!_jobFinished) {
    this.wait()
  }
  attachTry match {
    case _: Success[_] => detachSigintHandler()
    case _: Failure[_] => // Ignore error. Signal handler is on a best effort basis.
  }
```
Hopefully it is more clear now.